pkg/adapter/v2/configurator_configmap.go

```diff
@@ -231,10 +230,9 @@ func (c *tracingConfiguratorFromConfigMap) SetupTracing(ctx context.Context, cfg
 	logger := logging.FromContext(ctx)
 
 	cmw := ConfigWatcherFromContext(ctx)
-	service := fmt.Sprintf("%s.%s", cfg.InstanceName, NamespaceFromContext(ctx))
 
 	logger.Infof("Adding Watcher on ConfigMap %s for tracing", c.configMapName)
-	if err := tracing.SetupDynamicPublishing(logger, cmw, service, c.configMapName); err != nil {
+	if err := tracing.SetupDynamicPublishing(logger, cmw, "", c.configMapName); err != nil {
```

Comment:
since when `serviceName == ""` zipkin library will default to use `os.Hostname()`, I'm wondering if this will cause spans to report weird names for the service name and so traces won't be easily queriable.

what if we use this approach:
```
diff --git a/cmd/mtping/main.go b/cmd/mtping/main.go
index 9b0905bc3..7268f13b9 100644
--- a/cmd/mtping/main.go
+++ b/cmd/mtping/main.go
@@ -46,7 +46,7 @@ func main() {
        ctx = adapter.WithConfiguratorOptions(ctx, []adapter.ConfiguratorOption{
                adapter.WithLoggerConfigurator(adapter.NewLoggerConfiguratorFromConfigMap(component)),
                adapter.WithMetricsExporterConfigurator(adapter.NewMetricsExporterConfiguratorFromConfigMap(component)),
-               adapter.WithTracingConfigurator(adapter.NewTracingConfiguratorFromConfigMap()),
+               adapter.WithTracingConfigurator(adapter.NewTracingConfiguratorFromConfigMap(adapter.WithTracingConfiguratorServiceName(""))),
                adapter.WithProfilerConfigurator(adapter.NewProfilerConfiguratorFromConfigMap()),
                adapter.WithCloudEventsStatusReporterConfigurator(adapter.NewCloudEventsReporterConfiguratorFromConfigMap()),
        })
diff --git a/pkg/adapter/v2/configurator_configmap.go b/pkg/adapter/v2/configurator_configmap.go
index 6c6b1964f..b62f77cfb 100644
--- a/pkg/adapter/v2/configurator_configmap.go
+++ b/pkg/adapter/v2/configurator_configmap.go
@@ -200,6 +200,7 @@ func (c *metricsExporterConfiguratorFromConfigMap) SetupMetricsExporter(ctx cont
 // configures tracing using a watcher on a Configmap.
 type tracingConfiguratorFromConfigMap struct {
        configMapName string
+       serviceName   string
 }
 
 // TracingConfiguratorFromConfigMapOption for teawking the tracing configurator.
@@ -212,6 +213,13 @@ func WithTracingConfiguratorConfigMapName(name string) TracingConfiguratorFromCo
        }
 }
 
+// WithTracingConfiguratorServiceName sets the Service name for the tracing configuration.
+func WithTracingConfiguratorServiceName(name string) TracingConfiguratorFromConfigMapOption {
+       return func(c *tracingConfiguratorFromConfigMap) {
+               c.serviceName = name
+       }
+}
+
 // NewTracingConfiguratorFromConfigMap returns a ConfigMap based tracing configurator.
 func NewTracingConfiguratorFromConfigMap(opts ...TracingConfiguratorFromConfigMapOption) TracingConfigurator {
        c := &tracingConfiguratorFromConfigMap{
@@ -231,7 +239,10 @@ func (c *tracingConfiguratorFromConfigMap) SetupTracing(ctx context.Context, cfg
        logger := logging.FromContext(ctx)
 
        cmw := ConfigWatcherFromContext(ctx)
-       service := fmt.Sprintf("%s.%s", cfg.InstanceName, NamespaceFromContext(ctx))
+       service := c.serviceName
+       if service == "" {
+               service = fmt.Sprintf("%s.%s", cfg.InstanceName, NamespaceFromContext(ctx))
+       }
 
        logger.Infof("Adding Watcher on ConfigMap %s for tracing", c.configMapName)
        if err := tracing.SetupDynamicPublishing(logger, cmw, service, c.configMapName); err != nil {
```

cc @odacremolbap since you worked in this area recently.